test/hotspot/jtreg/compiler/lib/compile_framework/CompileFramework.java

```diff
@@ -0,0 +1,254 @@
+        compileJavaSources(sourceDir, javaSources);
+        setUpClassLoader();
+    }
+
+    private static String getSourceDirName() {
+        // Create temporary directory for jasm source files
+        final String sourceDir;
+        try {
+            sourceDir = "compile-framework-sources-" + ProcessTools.getProcessId();
+        } catch (Exception e) {
+            throw new InternalCompileFrameworkException("Could not get ProcessID", e);
+        }
+        System.out.println("Source directory: " + sourceDir);
+        return sourceDir;
+    }
+
+    private static void compileJasmSources(String sourceDir, List<SourceCode> jasmSources) {
+        if (jasmSources.size() == 0) {
+            System.out.println("No jasm sources to compile.");
+            return;
+        }
+        System.out.println("Compiling jasm sources: " + jasmSources.size());
+
+        List<String> jasmFileNames = writeSourcesToFile(sourceDir, jasmSources);
+        compileJasmFiles(jasmFileNames);
+        System.out.println("Jasm sources compiled.");
+    }
+
+    private static void compileJasmFiles(List<String> fileNames) {
+        // Compile JASM files with asmtools.jar, shipped with jtreg.
+        List<String> command = new ArrayList<>();
+
+        command.add("%s/bin/java".formatted(System.getProperty("compile.jdk")));
+        command.add("-classpath");
+        command.add(getAsmToolsPath());
+        command.add("org.openjdk.asmtools.jasm.Main");
+        command.add("-d");
+        command.add(System.getProperty("test.classes"));
+        for (String fileName : fileNames) {
+            command.add(new File(fileName).getAbsolutePath());
+        }
+
+        executeCompileCommand(command);
+    }
+
+    private static String[] getClassPaths() {
+        String separator = File.pathSeparator;
+        return System.getProperty("java.class.path").split(separator);
+    }
+
+    private static String getAsmToolsPath() {
+        for (String path : getClassPaths()) {
+            System.out.println("jtreg.jar?: " + path);
+            if (path.endsWith("jtreg.jar")) {
+                File jtreg = new File(path);
+                File dir = jtreg.getAbsoluteFile().getParentFile();
+                File asmtools = new File(dir, "asmtools.jar");
+                if (!asmtools.exists()) {
+                    throw new InternalCompileFrameworkException("Found jtreg.jar in classpath, but could not find asmtools.jar");
+                }
+                return asmtools.getAbsolutePath();
+            }
+        }
+        throw new InternalCompileFrameworkException("Could not find asmtools because could not find jtreg.jar in classpath");
+    }
+
+    private static void compileJavaSources(String sourceDir, List<SourceCode> javaSources) {
+        if (javaSources.size() == 0) {
+            System.out.println("No java sources to compile.");
+            return;
+        }
+        System.out.println("Compiling Java sources: " + javaSources.size());
```

Comment:
Same wish to make debug printouts controllable.
A dedicated (mini-?)logger, a static on/off variable, a dedicated PrintStream the user could set up themselves, etc.

Reply:
I think the flag will be off by default, so no printing. But if there ever arises a need for a logger like that, we can make this happen in a follow-up RFE.